Src/ILGPU.Tests/ArrayViews.cs

```diff
@@ -576,6 +576,54 @@ public void ArrayViewMultidimensionalAccess(long length)
             Verify(buffer, expectedData);
         }
 
+        internal static void ArrayViewSparseMultidimensionalAccessKernel(
+            Index1 index,
+            ArrayView<byte, LongIndex1> data,
+            ArrayView<byte, LongIndex3> source)
+        {
+            var reconstructedIndex = source.Extent.ReconstructIndex(
+                index.Size * (source.Extent.Size / data.Extent.Size));
+            data[index] = source[reconstructedIndex];
+        }
+
+        [Theory]
+        [InlineData(127)]
+        [InlineData(1300)]  // Makes a 2.04GB MemoryBuffer on the GPU.
```

Comment:
Although I like the idea of allocating a larger buffer on the GPU, I am afraid that this can easily lead to serious issues in the CI and on several developer machines. Since multiple test instances will run in parallel, this can easily cause OutOfMemory exceptions and unexpected test crashes. Maybe we should consider doing "dry runs" instead that just simulate a long array view that will be converted into a multidimensional one? (e.g. using the `ArrayView(ArrayViewObject, long index, long extent)` consructor)

Reply:
Sorry again for not getting back to you sooner on this PR 😞.

Reply:
That's OK! I'm glad we're on it together now.

I share your concern about the test as written. I think I understand what you're suggesting, though as I understand it (please correct me if I'm wrong), the test you're proposing would mostly just test whether it's possible to _construct_ a large ArrayView2D, and not whether it's possible to use it. It's not clear to me that such a test adds much value, since the only thing it's going to check for is that nobody put in a new spurious size check in code called during a constructor. I think it's more interesting to check that nothing is broken end-to-end (which is now done), and that nothing _will become_ broken end-to-end, which I worry isn't possible if we can't actually work with over 2GB of memory. I don't have a good solution.

Reply:
Yes, I totally agree that such a test may be of no use at all 🤔 Maybe we can work around this problem by allocating a much smaller buffer, adjusting the base pointer to point to `basePointer - 2GB` in GPU memory, and try to create an `ArrayView` instance (bypassing the existing assertions) to simulate a much larger view in order to index elements above the `2GB` boundary. However, I am not sure how to implement such a test without too much effort....

Reply:
Yeah, I guess testing construction and indexing is better than nothing, but I share your concerns about effort... and I worry that it might spuriously fail in other situations when very reasonable assertions are added.